sql/src/test/java/io/crate/integrationtests/DiskUsagesITest.java

```diff
@@ -0,0 +1,139 @@
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+import static java.util.stream.Collectors.toList;
+import static org.elasticsearch.cluster.routing.allocation.DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_REROUTE_INTERVAL_SETTING;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class DiskUsagesITest extends SQLTransportIntegrationTest {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        ArrayList<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins());
+        // Use the mock internal cluster info service, which has fake-able disk usages
+        plugins.add(MockInternalClusterInfoService.TestPlugin.class);
+        return plugins;
+    }
+
+    public void testRerouteOccursOnDiskPassingHighWatermark() throws Exception {
+        for (int i = 0; i < 3; i++) {
+            // ensure that each node has a single data path
+            internalCluster().startNode(Settings.builder().put(Environment.PATH_DATA_SETTING.getKey(),
+                                                               createTempDir()));
+        }
+
+        List<String> nodeIds = StreamSupport.stream(clusterState().getRoutingNodes().spliterator(), false)
+            .map(RoutingNode::nodeId)
+            .collect(toList());
+
+        // Start with all nodes at 50% usage
+        var clusterInfoService = (MockInternalClusterInfoService)
+            internalCluster().getInstance(ClusterInfoService.class, internalCluster().getMasterName());
+        clusterInfoService.setUpdateFrequency(TimeValue.timeValueMillis(200));
+        clusterInfoService.onMaster();
+        clusterInfoService.setN1Usage(nodeIds.get(0), new DiskUsage(nodeIds.get(0), "n1", "/dev/null", 100, 50));
+        clusterInfoService.setN2Usage(nodeIds.get(1), new DiskUsage(nodeIds.get(1), "n2", "/dev/null", 100, 50));
+        clusterInfoService.setN3Usage(nodeIds.get(2), new DiskUsage(nodeIds.get(2), "n3", "/dev/null", 100, 50));
+
+        execute("SET GLOBAL TRANSIENT" +
+                "   cluster.routing.allocation.disk.watermark.low='80%'," +
+                "   cluster.routing.allocation.disk.watermark.high='90%'," +
+                "   cluster.routing.allocation.disk.watermark.flood_stage='100%'");
+        // cluster.routing.allocation.disk.reroute_interval isn't exposed setting
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setTransientSettings(
+            Settings.builder().put(CLUSTER_ROUTING_ALLOCATION_REROUTE_INTERVAL_SETTING.getKey(), "1ms")));
```

Comment:
couldn't this be done via `nodeSettings()`?

Reply:
thanks! Yep, it can be done like that. I'll do it per test.